AccountRepository.cls

```diff
@@ -0,0 +1,20 @@
+public class AccountRepository {
+    
+	public static Account findOneAccountByNameLike(String searchName) {
+        List<Account> accounts = [SELECT Id, Name FROM Account WHERE Name LIKE :('%' + searchName + '%') LIMIT 1];
+        if (accounts.size() > 0) {
+            Account oneAccount = accounts.get(0);
+            return oneAccount;
+        }
+        return new Account();
+    }
+    
+    public static List<Account> findAllAccounts() {
+        List<Account> accounts = [SELECT Id, Name FROM Account];
```

Comment:
Always some limit --> such query can hit limit if there are more than 50000 records in database